### PR TITLE
Added smart action to perform depending on the cursor context (#329)

### PR DESCRIPTION
**Pull request by CaeChao**
### Added

- Added a smart action to perform depending on the context (textobject the cursor is on).

### Changed

* Added custom UI opts for toggle checkbox utils, and changed its behavior to cycle through checkboxes

### CODE REVIEW

File: lua/obsidian/commands/checkbox.lua [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+---@param client obsidian.Client
+return function(client)
+  local checkboxes = vim.tbl_keys(client.opts.ui.checkboxes)
+  for k, v in pairs(client.opts.ui.checkboxes) do
+    local order = v.order
+    if order and type(order) == "number" then
+      -- sort the checkboxes based on order
+      checkboxes[order] = k
+    end
+  end
```
Comment (epwalsh) [inline lua/obsidian/commands/checkbox.lua:12]: I think sorting here based on order would be more robust. Something like:

```suggestion
  table.sort(checkboxes, function(a, b)
    return client.opts.ui.checkboxes[a].order < client.opts.ui.checkboxes[b].order
  end)
```

**Review by epwalsh — CHANGES_REQUESTED** [review]
Thanks for this @CaeChao! Just a few comments

File: lua/obsidian/commands/checkbox.lua (hunk 1 above)
Comment (CaeChao) [reply lua/obsidian/commands/checkbox.lua]: Thanks for the advice! I'll commit the changes

**Comment by sotte** [comment]
Not sure if we want to add something like this, but accessing the tags via the "smart action" is quite handy IMO:

```lua
  -- show tags if possible
  local current_wrod = vim.fn.expand("<cWORD>")
  if current_wrod:match("^#") then
    vim.cmd("ObsidianTags")
    return
  end
```

@CaeChao thanks for bringing this feature over the finish line.

**Pull request merged**